formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Encoding.kt

```diff
@@ -290,7 +351,14 @@ internal open class CborReader(private val cbor: Cbor, protected val decoder: Cb
             knownIndex
         } else {
             if (isDone()) return CompositeDecoder.DECODE_DONE
-            val (elemName, tags) = decoder.nextTaggedString()
+            val (elemName, tags) = runCatching {
+                decoder.nextTaggedString()
+            }.getOrElse {
+                val serialLabel = decoder.nextNumber(null)
+                val elemName = descriptor.getElementNameForSerialLabel(serialLabel)
+                    ?: throw CborDecodingException("SerialLabel unknown: $serialLabel")
+                elemName to ulongArrayOf() // TODO Tags?
```

Comment:
yes, please! a function `nextTaggedNumber` should work like `nextTaggedString`, right?!

Reply:
Good point, that did catch an error in my implementation